code/model/modelanimation_segments.cpp

```diff
@@ -1213,37 +1217,73 @@ namespace animation {
 		m_segment->exchangeSubmodelPointers(replaceWith);
 	}
 
-	void ModelAnimationSegmentSoundDuring::playStartSnd(int pmi_id) {
-		if (snd_is_playing(m_instances[pmi_id].currentlyPlaying))
-			snd_stop(m_instances[pmi_id].currentlyPlaying);
+	void ModelAnimationSegmentSoundDuring::playStartSnd(polymodel_instance* pmi) {
+		if (snd_is_playing(m_instances[pmi->id].currentlyPlaying))
+			snd_stop(m_instances[pmi->id].currentlyPlaying);
 		
 		if(m_start.isValid())
-			m_instances[pmi_id].currentlyPlaying = snd_play(gamesnd_get_game_sound(m_start));
+			m_instances[pmi->id].currentlyPlaying = playSnd(pmi, m_start, false);
 	}
-	void ModelAnimationSegmentSoundDuring::playEndSnd(int pmi_id) {
-		if (snd_is_playing(m_instances[pmi_id].currentlyPlaying))
-			snd_stop(m_instances[pmi_id].currentlyPlaying);
+	void ModelAnimationSegmentSoundDuring::playEndSnd(polymodel_instance* pmi) {
+		if (snd_is_playing(m_instances[pmi->id].currentlyPlaying))
+			snd_stop(m_instances[pmi->id].currentlyPlaying);
 
 		if (m_end.isValid()) 
-			m_instances[pmi_id].currentlyPlaying = snd_play(gamesnd_get_game_sound(m_end));
+			m_instances[pmi->id].currentlyPlaying = playSnd(pmi, m_end, false);
+	}
+
+	sound_handle ModelAnimationSegmentSoundDuring::playSnd(polymodel_instance* pmi, const gamesnd_id& sound, bool loop) {
+		if (m_submodel != nullptr && pmi->objnum > 0) {
```

Comment:
Should this be >= ?

Reply:
As per object.h:
``int Object_next_signature = 1;	//0 is bogus, start at 1``
Similarly, the scripting API also considers objnum == 0 to be invalid.

Hence, I'd say that checking against > 0 in correct, but don't ask my why it was made that way.

Reply:
right. Object _signatures_ are always > 0, but objnums can be 0, I think